lib/selections.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * @typedef {Object} MinimalSelectionItem
+ * @property {number} from
+ * @property {number} to
+ */
+
+/** A selection of pieces to download.
+ * @typedef {MinimalSelectionItem & {
+ *  offset: number,
+ *  priority: number,
+ *  notify: function,
+ *  isStreamSelection?: boolean
+ * }} SelectionItem
+ */
+
+export class Selections {
+  /**
+   * @type {Array<SelectionItem>}
+   */
+  _items = []
+
+  /**
+   * @param {MinimalSelectionItem & {isStreamSelection: boolean, priority?: number}} item Interval to be removed from the selection
+   */
+  remove (item) {
+    for (let i = 0; i < this._items.length; i++) {
+      const existing = this._items[i]
+
+      if (existing.isStreamSelection) {
+        if (item.isStreamSelection) {
+          // If both are stream selections and they match, then we remove the first matching item, then we break the loop
+          if (existing.from === item.from && existing.to === item.to && existing.priority === item.priority) {
```

Comment:
i was stupid, this priority check isnt needed

Reply:
Are you sure? This check is for stream selections only. Those are still on the old logic: from, to and prio have to match and if you have multiple selections for the same range, then you have to call deselect for each.

Reply:
but stream selections cannot be created with priorities, unless some1 does it manually ,which isn't it's intended use case? or am I overlooking smth

Reply:
sorry i mean the if statement stays, but the priority check is useless

Reply:
Oh, okay. I didn't think it through and thought there might be stream selections with differing priorities, but that's not the case. In that case, I removed the priority comparison from the if statement.

Reply:
ah no, streams simply select needed data with the default unset priority, and data that's needed for the stream load RIGHT NOW is set to critical, which is highest priority